src/tests/unit/connection.test.js

```diff
@@ -0,0 +1,80 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import config from 'config';
+import _ from 'lodash';
+import proxyquireModule from 'proxyquire';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+
+// Prevent call thru to original dependencies
+const proxyquire = proxyquireModule.noCallThru();
+
+chai.should();
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+afterEach(() => { sinon.restore(); });
+
+describe('Test oracledb connection module', () => {
+  let configGetStub;
+  let connection;
+
+  beforeEach(() => {
+    configGetStub = sinon.stub(config, 'get')
+      .withArgs('dataSources.oracledb')
+      .returns({});
+  });
+
+  const createOracleDbStub = (createPoolStub) => {
+    connection = proxyquire('api/v1/db/oracledb/connection', {
+      config: { get: configGetStub },
+      oracledb: { createPool: createPoolStub },
+      // suppress logger output for testing
+      '../../../../utils/logger': { logger: { error: () => {} } },
+    });
+  };
+
+  describe('getConnection', () => {
+    it('Should call createPool if pool is falsy. Should not call createPool additional times', async () => {
+      const createPoolStub = sinon.stub()
+        .resolves({ getConnection: async () => 'test-connection' });
+      createOracleDbStub(createPoolStub);
+      const firstResult = connection.getConnection();
+      await firstResult.should.eventually.be.fulfilled.and.deep.equal('test-connection');
+      const secondResult = connection.getConnection();
+      await secondResult.should.eventually.be.fulfilled.and.deep.equal('test-connection');
```

Comment:
Looks like the logic is repeated, I guess we could use a loop to improve it. Also, I think `test-connection` should be stored in a variable. Can you also add a note saying why we need to test it twice?